### ISSUE TITLE
Is there a reliable way to set skipLoadPopstate = true when my app loads in?

### ISSUE THREAD

**Issue by stsquared99** [liferay/liferay-frontend-projects#834]
### Issue type (mark with `x`)

-   [ ] :thinking: Question
-   [ ] :bug: Bug report
-   [x] :gift: Feature request
-   [ ] :woman_shrugging: Other

### Description

**Desired behavior:**

https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/app/App.js#L1085
^ Is there a way to prevent skipLoadPopState being set to true here? 

**Current behavior:**
My app is making use of the History api and is binding to the popstate event, and senna's behavior visually interferes with that by doing a senna refresh on every popstate. Currently, the above line is the last thing that fires when I do a full page reload, so I currently don't have a way to set skipLoadPopState to true when my app loads in, it will always get overwritten to false when senna fully loads in. I haven't found another event to bind to that fires after this one.

**Repro instructions (if applicable):**

**Other information (environment, versions etc):**

**Comment by bryceosterhaus** [comment]
Thanks for creating this. I wish I had time to be able to look into it, but unfortunately might not be able to get to this for a little while. @kresimir-coko if you are free, check this one out and see if we can come up with a proper way to handle this.

**Comment by kresimir-coko** [comment]
Added to my to-do list 😉

**Comment by kresimir-coko** [comment]
Hey @stsquared99 could you please provide us with instructions for reproducing this issue? 

I personally have never worked with Senna so having steps to follow to get to the state where I can test this issue would be great. 

One other thing to note (@bryceosterhaus for you too), is that Senna is under our `maintenance` projects which means that we don't plan on adding any features to the project, only fixing bugs.

**Comment by bryceosterhaus** [comment]
@kresimir-coko yeah I know it is under maintenance. This issue is more about helping provide an example of how to use both the native history API and listen to popstate without having senna interfere with a refresh.

Ideally we should be able to provide an example on the liferay-portal side of setting `skipLoadPopState` to true without having senna overwrite that. So in order to investigate, you will need to run liferay-portal. I would try adding this module `modules/apps/frontend-js/frontend-js-clay-sample-web` and then setup some sort of app that mimics the interaction that Sam is looking for.

**Comment by kresimir-coko** [comment]
Hey @stsquared99 here's my attempt at replicating your problem.

I have added a sample app to `liferay-portal` and deployed it (on the left in the image), I have also added a `console.log` to the [js-spa-web/App.js@onLoad_](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/app/App.js#L1078-L1091) function, to see how it behaves.
<img width="1534" alt="image" src="https://user-images.githubusercontent.com/24564752/154248907-1f7c3767-3773-413f-9b22-5e5203ede3d0.png">

On the left side of the image I check the value of `skipLoadPopstate` before and after I change it, and then the last log that happens is the one from the right (inside the js-spa-web/App.js@onLoad_) confirming the value has been properly set in the file on the left.

Basically, I have managed to override the value of `skipLoadPopstate` in my own app by using `Liferay.SPA.app.skipLoadPopstate = true;`

Let me know if this is the problem you're facing or provide further details.
